lib/cn/csched_sp3.c

```diff
@@ -850,26 +851,20 @@ sp3_node_unlink(struct sp3 *sp, struct sp3_node *spn)
         sp3_rb_erase(sp->rbt + tx, spn->spn_rbe + tx);
 }
 
-/* Remove all nodes from all rb trees that belong to given cn_tree */
 static void
-sp3_node_unlink_all(struct sp3 *sp, struct cn_tree *tree)
+sp3_node_unlink_all(struct sp3 *sp, struct sp3_node *spn)
 {
-    struct cn_tree_node *tn;
-
-    cn_tree_foreach_node(tn, tree) {
-        struct sp3_node *spn = tn2spn(tn);
-
-        sp3_node_unlink(sp, spn);
-
-        list_del_init(&spn->spn_rlink);
-        list_del_init(&spn->spn_alink);
-    }
```

Comment:
This function was called from only one place, and messed me up bad when I tried to use it.  So I re-purposed it to remove a single node from all the lists, and move it node loop to the call site of the original caller.